t/lib/WTSI/NPG/iRODS/PublisherTest.pm

```diff
@@ -471,4 +473,44 @@ sub pf_stale_md5_cache {
      'Stale MD5 was regenerated') or diag explain $obj->metadata;
 }
 
+sub pf_md5_cache_ro {
+  my ($irods, $data_path, $coll_path) = @_;
+
+  my $cache_threshold = 1; # Create MD5 cache for files of 1 byte or
+                           # more
+  my $publisher = WTSI::NPG::iRODS::Publisher->new
+    (irods                    => $irods,
+     checksum_cache_threshold => $cache_threshold);
+
+  my $local_path_d = "$data_path/publish_file/d.txt";
+  my $remote_path = "$coll_path/pf_md5_cache_ro.txt";
+
+  try {
+    chmod 0555, "$data_path/publish_file/";
```

Comment:
(!-w "$data_path/publish_file/") or croak "Failed to make $data_path/publish_file/ not writable";
or similar to ensure that the change took place.

Reply:
Thanks - I meant to go back and put that in, but forgot.